### PR TITLE
Revert "Rad collectors gone, tesla coils to generate power"

### PR DESCRIPTION
**Pull request by MrsTonedOne**
Requested

don't worry about #61917 getting locked, you can continue the discussion in here.

### CODE REVIEW

**Comment by esainane** [comment]
inb4 shitstorm

**Comment by Moltovz** [comment]
Woo

**Comment by InsaneRed** [comment]
seeing as how most of the critisim was just marked as spam on the original post and not read by maintainers (?) this is needed

**Review by Maurukas — APPROVED** [review]

**Comment by Iamgoofball** [comment]
@InsaneRed There was exactly 1 comment marked as spam on that PR.

It was a brigading post from our downstream, Skyrat. They posted that comment 5 minutes after the call to action for a brigade was posted in their discord.

That user has literally 1 connection to /tg/, and they didn't even finish connecting because it returns zero rounds when you click their name in scrubby.
chrome_srnantnMb9
chrome_CMbPyEFt21

**Comment by optimumtact** [comment]
You need to fill out a justification for your PR.

The other PR author at least went to that level of effort, nobody can seriously consider this one unless you do.

**Comment by Acer202** [comment]
I can't believe MSO just pulled a Charlie Brown Christmas on us with radiation collectors
Yoink

**Comment by Ghilker** [comment]
you're an idiot

**Comment by esainane** [comment]
> tesla bolts also have a tendency to straight up jump to players if they aren't wearing insuls, and they're out of the effective range of grounding rods

There's no such thing as "being outside of effective range of grounding rods" with the current setup.

All potential targets are considered. A player on a bicycle has the highest priority. A Tesla Coil has the second highest priority, but has a chance to be skipped. A Grounding Rod has the third highest priority, with no chance of being skipped.

If a bolt is arcing to a player instead of a Grounding Rod, they're either on a bicycle, went out of their way to build a long chain of tesla coils leading far outside of the engine, or someone unanchored the Grounding Rod - or failed to anchor it, if it was spawned in. Roundstart Grounding Rods are anchored by default.

I really recommend everyone read the PR or the code in question, since there is quite a lot of misinformation being spread around.

[The Radiation Collector code](https://github.com/tgstation/tgstation/blob/master/code/modules/power/singularity/collector.dm) still exists, and they still admin spawnable on /tg/ today. Downstreams are free to use them, place them on maps, and add recipes as they see fit with very little effort.

[The Supermatter Zap code]() follows very simple logic, and it's very easy to stay safe around them. Again, if the supermatter crystal is undergoing extreme delamination, AND the engine room underwent significant sabotage, you might need to make sure you have gloves and a hardsuit. This was true of the old Supermatter Crystal too, which still made overcharged bolts in extreme delamination situations. Nothing has changed here, except that Grounding Rods are available and installed at round start. These perfectly solve all lightning bolt problems, including the situations where they would have previously been a massive problem, unless sabotaged. If overcharged lightning bolts are the problem, they are strictly less of a problem than they were previously.

**Comment by tralezab** [comment]
i double checked with ghilker, i don't see anything crazy going on with zaps as you say yawet

the ranges on the views are very reasonable and wouldn't create too much lag. you gotta consider that ai datums are making bigger view calls in bigger numbers at roundstart than the sm would be

**Comment by Yawet330** [comment]
> > tesla bolts also have a tendency to straight up jump to players if they aren't wearing insuls, and they're out of the effective range of grounding rods
> 
> There's no such thing as "being outside of effective range of grounding rods" with the current setup.
> 
> All potential targets are considered. A player on a bicycle has the highest priority. A Tesla Coil has the second highest priority, but has a chance to be skipped. A Grounding Rod has the third highest priority, with no chance of being skipped.
> 
> If a bolt is arcing to a player instead of a Grounding Rod, they're either on a bicycle, went out of their way to build a long chain of tesla coils leading far outside of the engine, or someone unanchored the Grounding Rod - or failed to anchor it, if it was spawned in. Roundstart Grounding Rods are anchored by default.
> 
> I really recommend everyone read the PR or the code in question, since there is quite a lot of misinformation being spread around.
> 
> [The Radiation Collector code](https://github.com/tgstation/tgstation/blob/master/code/modules/power/singularity/collector.dm) still exists, and they still admin spawnable on /tg/ today. Downstreams are free to use them, place them on maps, and add recipes as they see fit with very little effort.
> 
> [The Supermatter Zap code](https://github.com/tgstation/tgstation/blob/master/code/modules/power/supermatter/supermatter.dm#L1153) follows very simple logic, and it's very easy to stay safe around them. Again, if the supermatter crystal is undergoing extreme delamination, AND the engine room underwent significant sabotage, you might need to make sure you have gloves and a hardsuit. This was true of the old Supermatter Crystal too, which still made overcharged bolts in extreme delamination situations. Nothing has changed here, except that Grounding Rods are available and installed at round start. These perfectly solve all lightning bolt problems, including the situations where they would have previously been a massive problem, unless sabotaged. If overcharged lightning bolts are the problem, they are strictly less of a problem than they were previously.

Said overcharged bolts did not husk or instantly explode; as far as I am aware, explosions only happened at truly high MeV's (in the 10-20KMEV ranges, but as for everything else; this is actually a good point, my main concern is what happens when the SM is modified to provide more power, while currently you are unable to be outside of the effective grounding range with a roundstart setup; engineers have a horrible tendency to min-max a high-efficiencty SM build, with the grounding rods usually placed closer, or actually where walls used to be, in order to save space and allow for some funny setups (like a turbine directly connected to the SM, abusing superpumps to generate 800kw extra power).

Said movement of the grounding rods also increases their chance of incidentally allowing sabotage to be easier, which is prevalent with events such as meteorites (as station shields are now an objective, versus always orderable, at least as far as I am aware), which can actually cause the above event to happen with ease.

Thirdly; I am also aware that grounding rods can only ground so many strikes at once in a small area, which I have seen actually occur when a cranked up SM running at 9000MeV on old SM code (with 2.7K carbon dioxide, abusing a heat exchanger bug for testing purposes), managed to bypass the four grounding rods literally right outside of the radiation collector, and cause an explosion in the nearby plasmaglass window (having arced to the grille), damaging the loop enough to have a stacking delam.

Again, my point is less that it's a poor mechanical change, infact, Tesla bolts can be /more/ fun due to the way their arcing system works, it's that it was done for the wrong reasons, and hasn't been correctly considered I believe.


TLDR: Good change, bad reasons for it, not too many considerations given for specific maps.

Also, Tralezab; this is actually true, my point wasn't that it would cause any actual severe amount of lag, more that it was technically more per tile than radiation code; although radiation could effect far greater areas, eventually causing a cut-off point.

**Comment by Ghilker** [comment]
@Yawet330 Man you never stop blabbering about things you don't understand aren't ya?

**Comment by Yawet330** [comment]
> @Yawet330 Man you never stop blabbering about things you don't understand aren't ya?

And yet most of my points are not based off of SM code and instead general overarching potential issues with the idea.

**Comment by esainane** [comment]
if your points aren't ground in /tg/ code or experiences on /tg/ servers I expect there to continue to be a lot of talking past one another

**Comment by esainane** [comment]
You're looking at tesla coils

tesla coils consume part of the charge and generate power from it

grounding rods consume all remaining charge and terminate it

**Comment by Yawet330** [comment]
Yep, my bad there.

**Comment by esainane** [comment]
like it's literally six lines of pretty unambiguous code

I assume skyrat has a repository somewhere since we're AGPL

so if you have a problem with some custom skyrat thing I suggest you open an issue or PR there

**Comment by WarlockD** [comment]
Everyone is glossing over what the real issue is with all of this.  

PEOPLE DON'T UNDERSTAND OR CARE ABOUT HOW DANGROUS RADATION IS

When I started playing I understood the differences between radiation sickness and exposure.  I even remember when it was discussed last year about removing it how agent I was of it.  It took playing the damn game for lord knows how many hours and talking with players to realize that it doesn't matter how well you explain it in the wiki that people still get misinformed.  Like throwing someone under water doesn't cure sickness, never did!  If your glory it will get rid of that, hell we even buffed that but you will still die of toxen damage.  The health scanners show the stupid percentages and meds works fine on people, again buffed.

I would be yelling at orange on removing it but now I am just mixed.  If we want to keep radiation, a simple fix is get rid of exposure and make sickness more "instant"  That is you start throwing up at even 400% so somone knows its wrong.  Mutations start at 600%.  Waves only appear from radiation sources (SM, Cobolt-60, banned X-ray machines?)

At the end of the day, this is a game mechanic.  Simplified, "This area is dangerous, you MUST wear a suit with some protection or die"

And just as a final note, how the fuck are people getting "attacked" by radiation?  The only real source is the SM and I keep hearing this argument but not seeing it in play unless idiots are playing with the damn reactor.

**Comment by Maurukas** [comment]
@WarlockD Should be listened to. His refactor of radiation was significant.

At this stage I am seriously annoyed by the merging of unreviewed map edits, the poor balance/trivial cheesability of this and the incredibly obnoxious sound.

**Comment by optimumtact** [comment]
stay in your lane maptainer

**Comment by LemonInTheDark** [comment]
> 
> 
> the code is extremely simple, so I don't see any places for mistakes on the /tg/ side
> 
> > Edit: I actually wonder, would it be possible for an arc to somehow be unable to path to a grounding rod or tesla coil, despite there being one nearby thinking
> 
> ```dm
> for(var/test in oview(zapstart, range))
> ```
> 
> not on /tg/

I have bad news
<https://github.com/tgstation/tgstation/issues/50504>

Also of note, zap chains keep track of who they've zapped, and how. they won't duplicate those zaps, or effectively double back.
See `ZAP_ALLOW_DUPLICATES`. Would only cause issues if someone (past me) fucked up with how that list is passed around. I don't think I did though.

I've seen some somewhat weird things in the past, so it might be worth looking into, but it's not intended that's for sure

**Comment by WitherVictor** [comment]
I can't understand, these changes are killing game features, I guess no one need another Among Us.

**Comment by aarrbba123** [comment]
I think that rad collectors should be an option for servers to use,like a config that tells the server to put either rad collectors or tesla coils.Either way,radiation needs some balancing or it will be removed

**Comment by aarrbba123** [comment]
> Absolutely not

Ahh well,nothing I can do

**Comment by WitherVictor** [comment]
> 
> 
> > > Sus dude
> > 
> > 
> > Sus for what bro? I'm telling facts. I play this game for it's massive content, you guys are making it less and less, why can't I turn toward games like Barotrauma for better game engine and art ? I guess you guys lost the goals of games developers should focus on.
> 
> Go play that game then why you still here

Dude say something instead of attacking others, I'm just telling truth. Just check videos about introducing SS13, all of them mentioned one thing, massive content, recently updates just killing game feature, so is there anything wrong about what I said ?

**Comment by Ghilker** [comment]
Yes, you're wrong. You understand nothing about game balance, coding features and maintaining them. 
Having a massive ocean the depth of a puddle means nothing.
Having a lot of options but none of them have meaning to them is nothing.
Removing existing features to expand on others and make them better is a boost in game quality.

Also the game's free and open source, if you want more content code it yourself

**Comment by Ghilker** [comment]
Also say something useful yourself, in your post you haven't said anything useful
You just say that this kill the game, massive features but nothing else

**Comment by EricZilla** [comment]
> > Wither is not doing tasks
> 
> Why I have to spend my time on a old Python engine game and arrogant coders ? Copy codes from Goon can make you arrogant like this ? Come on, no more jokes pls.

You gotta put in the work if you want your way, thats what the coders do, and thats why random people complaining about deadend features getting removed tend to be ignored, especially if they dont even actually propose any solutions
